graphql-java-support/src/main/java/com/apollographql/federation/graphqljava/FederationSdlPrinter.java

```diff
@@ -932,16 +932,18 @@ private void printComments(PrintWriter out, Object graphQLType, String prefix) {
             List<String> lines = Arrays.asList(descriptionText.split("\n"));
             if (options.isDescriptionsAsHashComments()) {
                 printMultiLineHashDescription(out, prefix, lines);
+            } else if (needsBlockString(descriptionText)) {
+                printMultiLineDescription(out, prefix, lines);
             } else {
-                if (lines.size() > 1) {
-                    printMultiLineDescription(out, prefix, lines);
-                } else {
-                    printSingleLineDescription(out, prefix, lines.get(0));
-                }
+                printSingleLineDescription(out, prefix, descriptionText);
             }
         }
     }
 
+    private boolean needsBlockString(String description) {
+        return description.contains("\"") || description.contains("\n") | description.contains("\r");
```

Comment:
According to the spec a backslash is also a character that can be used in Block Strings without escaping. At Indeed we use line feed and quote frequently, so this definitely hits the main ones we care about.

Reply:
See http://spec.graphql.org/draft/#sec-String-Value.Block-Strings